vapi/simulator/simulator.go

```diff
@@ -1876,6 +1876,23 @@ func (s *handler) libraryDeploy(ctx context.Context, c *vim25.Client, lib *libra
 		return nil, errors.New(spec.Error[0].LocalizedMessage)
 	}
 
+	if config != nil {
+		if vmImportSpec, ok := spec.ImportSpec.(*types.VirtualMachineImportSpec); ok {
+			var configSpecs []types.BaseVirtualDeviceConfigSpec
+
+			// Remove devices that we don't want to carry over from the import spec. Otherwise, since we
+			// just reconfigure the VM with the provided ConfigSpec later these devices won't be removed.
+			for _, d := range vmImportSpec.ConfigSpec.DeviceChange {
+				switch d.GetVirtualDeviceConfigSpec().Device.(type) {
+				case types.BaseVirtualEthernetCard:
```

Comment:
Yeah. I was thinking maybe we could ignore all devices in the Ovf ConfigSpec except Disk and Disk Controllers. But Since VM Op is the only consumer of this atm, just ignoring the VirtualEthernetCard should be enough in simulator land.

Reply:
I'll prb end up inverting the check then. I might merge the two ConfigSpecs too so we don't need to Reconfigure in here either but for what need out of the simulator that doesn't matter too much.